src/projection.jl

```diff
@@ -210,6 +210,9 @@ function (project::ProjectTo{AbstractArray})(dx::AbstractArray{S,M}) where {S,M}
     return dz
 end
 
+# Trivial case -- won't collapse Any[NoTangent(), NoTangent()]
+(project::ProjectTo{AbstractArray})(dx::AbstractArray{<:AbstractZero}) = NoTangent()
```

Comment:
Do we want:
```suggestion
(project::ProjectTo{<:AbstractArray})(dx::AbstractArray{<:AbstractZero}) = NoTangent()
```

So this applies also to other arrays?
I think we might

Reply:
Could do, currently it goes via the parent
```
julia> ProjectTo([1,2,3]')(fill(NoTangent(),1,3))
NoTangent()

julia> ProjectTo([1,2,3]')(fill(NoTangent(),50))
ERROR: DimensionMismatch("variable with size(x) == (1, 3) cannot have a gradient with size(dx) == (50,)")

julia> ProjectTo([1,2,3])(fill(NoTangent(),50)) # doesn't check size
NoTangent()
```

Reply:
Trying this leads me to ambiguities:
```
julia> @test padj([NoTangent() NoTangent() NoTangent()]) === NoTangent()
Error During Test at REPL[31]:1
  Test threw exception
  Expression: padj([NoTangent() NoTangent() NoTangent()]) === NoTangent()
  MethodError: (::ProjectTo{Adjoint, NamedTuple{(:parent,), Tuple{ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}}}}}}}})(::Matrix{NoTangent}) is ambiguous. Candidates:
    (project::ProjectTo{Adjoint})(dx::AbstractArray) in ChainRulesCore at /Users/me/.julia/dev/ChainRulesCore/src/projection.jl:259
    (project::ProjectTo{<:AbstractArray})(dx::AbstractArray{<:AbstractZero}) in ChainRulesCore at /Users/me/.julia/dev/ChainRulesCore/src/projection.jl:214
  Possible fix, define
    (::ProjectTo{Adjoint})(::AbstractArray{<:AbstractZero})
```